### PR TITLE
Add edit session functionality

### LINKED ISSUE #61: As a trainer, I can edit session details

### PR DESCRIPTION
**Pull request by maguireong**
Require some help in EditSessionCommandParserTest, 3 assertions in 1 of the method doesn't work quite right yet. Other than that, everything works fine.

### CODE REVIEW

**Comment by benclmnt** [comment]
Btw, perhaps it is better to test for checkstyle locally before pushing?

To run checkstyle and test cases, run `./gradlew check` from Intellij terminal

To edit your last commit (so as to clutter the commit history), do `git commit --amend --no-edit` (also in terminal)

File: src/test/java/seedu/address/testutil/TypicalSessions.java [hunk 1]
```diff
@@ -24,12 +26,14 @@
     public static final Session GETWELL = new SessionBuilder()
             .withGym(VALID_GYM_GETWELL)
             .withExerciseType(VALID_EXERCISE_TYPE_GETWELL)
-            .withInterval(VALID_START_TIME_GETWELL, VALID_DURATION_GETWELL)
+            .withInterval(LocalDateTime.parse(VALID_START_TIME_GETWELL, Interval.DATE_TIME_FORMATTER),
```
Comment (benclmnt) [inline src/test/java/seedu/address/testutil/TypicalSessions.java:29]: [open for discussion] Is it better to do the new way or the previous way? 

This change is related to changes in `SessionBuilder::withInterval`

cc @maguireong @dhafinrazaq

**Review by benclmnt — COMMENTED** [review]
There's a huge chunk commented out in `EditSessionCommandTest.java`

Are we going to need it? If not, perhaps it is better to remove it first and create a new Issue for this.

Other changes look sensible to me

**Comment by maguireong** [comment]
> There's a huge chunk commented out in `EditSessionCommandTest.java`
> 
> Are we going to need it? If not, perhaps it is better to remove it first and create a new Issue for this.
> 
> Other changes look sensible to me

That's for future tests because it requires the find session function for it to work, that's why I commented it out for now

File: src/test/java/seedu/address/testutil/TypicalSessions.java (hunk 1 above)
Comment (maguireong) [reply src/test/java/seedu/address/testutil/TypicalSessions.java:29]: I think both ways work but when I was doing it the other time, I didn't think much into it so I just put in the respective params into the method.

Comment (dhafinrazaq) [reply src/test/java/seedu/address/testutil/TypicalSessions.java:29]: in terms of consistency with the other with__ methods, should we pass the String? What do you all think?

**Pull request closed without merging**